jbmc/regression/jbmc-concurrency/synchronized-blocks/test4.desc

```diff
@@ -0,0 +1,9 @@
+CORE
+A.class
+--function 'A.aStatic' --lazy-methods
```

Comment:
This is missing `--java-threading` so the fact the core models aren't provided isn't relevant (so comment is wrong), I'd duplicate this test one with `--java-threading` and no models, one without `--java-threading` and models to check it is all working as expected.

Reply:
This is missing the  'java-threading'  flag on purpose. Java-threading option requires certain core models, but it is not the case that including the core models requires the  java-threading option. There other  pre-existing tests that  that specifically test that it works as expected with core-models (albeit not they are not testing for  exceptions). I will clarify the comment.

Reply:
So the options are:

 - no core modes and no --java-threading (Tested here)
 - core models and --java-threading (tested elsewhere)
 - --java-threading and no core models (invalid combination so no tests)
 - core models and no --java-threading  (I think this could benefit from a test to check nothing unexpected happens just because the symbol ` java::java.lang.Object.monitorexit:(Ljava/lang/Object;)V` is defined.